packages/medusa/src/api/routes/admin/product-categories/delete-products-batch.ts

```diff
@@ -0,0 +1,121 @@
+ * x-authenticated: true
+ * parameters:
+ *   - (path) id=* {string} The ID of the Product Category.
+ *   - (query) expand {string} (Comma separated) Category fields to be expanded in the response.
+ *   - (query) fields {string} (Comma separated) Category fields to be retrieved in the response.
+ * requestBody:
+ *   content:
+ *     application/json:
+ *       schema:
+ *         $ref: "#/components/schemas/AdminDeleteProductCategoriesCategoryProductsBatchReq"
+ * x-codegen:
+ *   method: removeProducts
+ * x-codeSamples:
+ *   - lang: Shell
+ *     label: cURL
+ *     source: |
+ *       curl --location --request DELETE 'https://medusa-url.com/admin/product-categories/{id}/products/batch' \
+ *       --header 'Authorization: Bearer {api_token}' \
+ *       --header 'Content-Type: application/json' \
+ *       --data-raw '{
+ *           "product_ids": [
+ *             {
+ *               "id": "{product_id}"
+ *             }
+ *           ]
+ *       }'
+ * security:
+ *   - api_token: []
+ *   - cookie_auth: []
+ * tags:
+ *   - Product Category
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           type: object
+ *           properties:
+ *             product_category:
+ *               $ref: "#/components/schemas/ProductCategory"
+ *   "400":
+ *     $ref: "#/components/responses/400_error"
+ *   "401":
+ *     $ref: "#/components/responses/unauthorized"
+ *   "404":
+ *     $ref: "#/components/responses/not_found_error"
+ *   "409":
+ *     $ref: "#/components/responses/invalid_state_error"
+ *   "422":
+ *     $ref: "#/components/responses/invalid_request_error"
+ *   "500":
+ *     $ref: "#/components/responses/500_error"
+ */
+
+export default async (req: Request, res: Response) => {
+  const { id } = req.params
+  const validatedBody =
+    req.validatedBody as AdminDeleteProductCategoriesCategoryProductsBatchReq
+
+  const productCategoryService: ProductCategoryService = req.scope.resolve(
+    "productCategoryService"
+  )
+
+  const manager: EntityManager = req.scope.resolve("manager")
+  await manager.transaction(async (manager) => {
+    return await productCategoryService.withTransaction(manager).removeProducts(
+      id,
+      validatedBody.product_ids.map((p) => p.id)
+    )
+  })
+
+  const productCategory = await productCategoryService.retrieve(
+    id,
+    req.retrieveConfig
+  )
+
+  res.status(200).json({ product_category: productCategory })
+}
+
+/**
+ * @schema AdminDeleteProductCategoriesCategoryProductsBatchReq
+ * type: object
+ * required:
+ *   - product_ids
+ * properties:
+ *   product_ids:
+ *     description: The IDs of the products to delete from the Product Category.
+ *     type: array
+ *     items:
+ *       type: object
+ *       required:
+ *         - id
+ *       properties:
+ *         id:
+ *           description: The ID of a product
+ *           type: string
```

Comment:
thought(non-blocking): I would've expected `product_ids` to be an array of string. I find it odd to have to pass the following in order to remove products by id:
`[{ id: "foo" }, { id: "bar" }]`
The following seems more convenient:
`["foo", "bar"]`

Reply:
We are all agree but this is a pattern introduced by seb with the idea to be able to pass some metadata in the future if needed. Some sort of configuration for the batches. In the mean tile it is a pattern applied a bit every where but not everywhere 😅 but yeah it seams odd and maybe should be named entity_data or something to avoid miss leading

Reply:
This has been added to the issue here - https://linear.app/medusajs/issue/CORE-1060/create-a-uniform-pattern-for-batch-operations

Goal of this issue is to come to a conclusion on how to approach these things, form a consensus and make these changes holistically, since there are a few places this comment would apply to. I stuck with the object to not introduce a deviation in existing pattern.

Reply:
Got it. Thanks for the explanation. LGTM.